### PR TITLE
Editing description of vulnerability

### PR DESCRIPTION
**Pull request by Cheshiriks**
Снимок экрана 2023-08-21 в 11 01 38
Снимок экрана 2023-08-21 в 11 02 47

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt [hunk 1]
```diff
@@ -63,14 +89,54 @@ val vulnerabilityGeneralInfo: FC<VulnerabilityGeneralInfo> = FC { props ->
...
-                    className = ClassName("font-weight-bold text-primary mb-4")
-                    +"Description"
+                div {
+                    className = ClassName("row")
+                    h6 {
+                        className = ClassName("ml-3 font-weight-bold text-primary mb-4")
+                        +"Description"
+                    }
+                    if (true) {
+                        ReactHTML.button {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt:99]:
```suggestion
                        button {
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt [hunk 2]
```diff
@@ -63,14 +89,54 @@ val vulnerabilityGeneralInfo: FC<VulnerabilityGeneralInfo> = FC { props ->
...
-                h6 {
-                    className = ClassName("font-weight-bold text-primary mb-4")
-                    +"Description"
+                div {
+                    className = ClassName("row")
+                    h6 {
+                        className = ClassName("ml-3 font-weight-bold text-primary mb-4")
+                        +"Description"
+                    }
+                    if (true) {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt:98]: Not good

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt [hunk 3]
```diff
@@ -63,14 +89,54 @@ val vulnerabilityGeneralInfo: FC<VulnerabilityGeneralInfo> = FC { props ->
...
+                    textarea {
+                        className = ClassName("auto_height form-control-plaintext pt-0 pb-0")
+                        value = description
+                        disabled = isEditDisabled
+                        rows = 8
+                        onChange = {
+                            setDescription(it.target.value)
+                        }
+                    }
+                    button {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt:120]: Why not `buttonBuilder`? Why don't you use it?

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt [hunk 4]
```diff
@@ -16,14 +22,34 @@ import react.dom.html.ReactHTML.textarea
...
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
 
 /**
  * [FC] that is used to display some general vulnerability information
  */
 @Suppress("EMPTY_BLOCK_STRUCTURE_ERROR", "MAGIC_NUMBER")
 val vulnerabilityGeneralInfo: FC<VulnerabilityGeneralInfo> = FC { props ->
+
+    val (isEditDisabled, setIsEditDisabled) = useState(true)
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt:37]: I would do the same as we did for settings: create a data class where you will change several fields, not only this one. And will have one single flag for it.

Right now for each and every change: name, short description, long description, you will create a state.
It's complicated and strange, let's make at least this thing properly

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt (hunk 2 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt:98]: Omg

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt]: ctrl+c / ctrl+v is evil :)
Fixed it

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt (hunk 4 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt:31]: Moved the Change button to the very top. The entire vulnerability is now transferred to the state. This is much easier than a separate date class.

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt [hunk 5]
```diff
@@ -16,14 +22,34 @@ import react.dom.html.ReactHTML.textarea
...
+    val (description, setDescription) = useStateFromProps(props.vulnerability.description.orEmpty())
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt]: But what is this then? Here you should have vuln. Object

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt [hunk 6]
```diff
@@ -18,21 +19,58 @@ import web.cssom.ClassName
...
+import kotlinx.serialization.json.Json
 
 /**
  * [FC] that is used to display some general vulnerability information
  */
 @Suppress("EMPTY_BLOCK_STRUCTURE_ERROR", "MAGIC_NUMBER")
 val vulnerabilityGeneralInfo: FC<VulnerabilityGeneralInfo> = FC { props ->
+
+    val (isEditDisabled, setIsEditDisabled) = useState(true)
+    val (vulnerability, setVulnerability) = useStateFromProps(props.vulnerability)
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfo.kt:32]: Good

**Pull request merged**